hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/records/GasFeeRecordBuilderTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.test.records;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.mock;
+
+import com.hedera.node.app.service.contract.impl.records.GasFeeRecordBuilder;
+import org.junit.jupiter.api.Test;
+
+class GasFeeRecordBuilderTest {
+    @Test
+    void withGasFeeWorksAsExpected() {
+        final var subject = mock(GasFeeRecordBuilder.class);
+        doCallRealMethod().when(subject).withGasFee(123L);
+        given(subject.transactionFee()).willReturn(456L);
+        given(subject.transactionFee(123L + 456L)).willReturn(subject);
```

Comment:
Really, isn't this use of a mock rather ... hmm ... well, not sure of the proper derogatory word here ... but this is an _interface_ where you're testing a _default method_ ... rather than this interesting mocking can't you just create an anonymous class inline here with the transaction fee set and then just call it?   In my mind that would be clearer than this 4 line setup of a mock.  

But as you know, I don't like mocks at all, so YMMV.

Reply:
Will replace with anonymous class in next PR. 😁